### PR TITLE
Asset|Document Listing getCount with Limit and Offset always returns 0

### LINKED ISSUE #5639: Asset/Listing getCount with Limit and Offset always returns 0
# Bug Report

### Expected behavior
Asset\Listing() ... $assetListing->getCount() with set Limit/Offset values should return correct assets count.

### Actual behavior
Asset\Listing() getCount() with Limit/Offset returns 0

### Steps to reproduce
- Create Asset\Listing instance
- Set Limit and Offset values
- Load list
- call getCount() on asset listing instance

### Code example
```php
$assetList = new Asset\Listing();
$assetList->setOffset(5);
$assetList->setLimit(10);

$assetList->load();
$x = $assetList->getCount();
// $x = 0
```

### Solution / additional informations
Not sure whats the expected solution, so no PR

If its a condition, that always before $myAssetList->getCount(), $myAssetList->load() must be called, then it just should return ``` count($this->model->getAssets())``` and not execute another query. 

If the current system with it's own query for getCount() has to stay this way, then the query in getCount() (models/Asset/Listing/Dao.php) should be reseted with:

```php
$select = $this->getQuery([new Expression('COUNT(*)')]);
$select->reset(QueryBuilder::LIMIT_COUNT);
$select->reset(QueryBuilder::LIMIT_OFFSET);
```

### PR DESCRIPTION
**Pull request by weisswurstkanone**
Resolves #5639

### CODE REVIEW

File: models/Asset/Listing/Dao.php [hunk 1]
```diff
@@ -99,10 +99,11 @@ public function loadIdList()
      */
     public function getCount()
     {
-        $select = $this->getQuery([new Expression('COUNT(*)')]);
-        $amount = (int) $this->db->fetchOne($select, $this->model->getConditionVariables(), $this->model->getConditionVariableTypes());
+        if (count($this->model->getAssets()) == 0) {
```
Comment (dpfaffenbauer) [inline models/Asset/Listing/Dao.php]: isn't that way slower? Cause it needs to hydrate the models.

Comment (weisswurstkanone) [reply models/Asset/Listing/Dao.php]: same approach as for objects. at least it produces the correct result

Comment (dpfaffenbauer) [reply models/Asset/Listing/Dao.php]: Why don't you use `getQuery` and change it up to make a COUNT(*) query?

https://github.com/pimcore/pimcore/blob/9167b96ed44aef0d5f029be73d120faec972f3a8/models/Asset/Listing/Dao.php#L64

Comment (dpfaffenbauer) [reply models/Asset/Listing/Dao.php]: just realised, that is exactly what has been proposed in #5639

**Pull request merged**